lib/assessment.js

```diff
@@ -8,38 +8,57 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
-
+var min = function min(previousValue, currentValue, index, array) {
+  return currentValue > previousValue ? currentValue : previousValue;
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-var sum = function sum() {
-
+var sum = function sum(previousValue, currentValue, index, array) {
+  return previousValue + currentValue;
 };
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
-var any = function any(array, callback, thisArg) {
 
+// this seems vague, and I don't exactly know
+// what's being asked here, so I'm making a function
+// that uses a for loop to set every item in the array
+// to be 1
+var any = function any(array, callback, thisArg) {
+  for (var i = 0; i < array; i++) {
+    callback(array[i], i, array) = 1;
+  }
+  return true;
 };
 
 // Question 4
 // Write a function that returns a new array
 // made up of the array elements for which invoking
 // the callback returns a truthy value
-var select = function select(array, callback, thisArg) {
 
+// Stil not clear. So, this is what I've got.
+var select = function select(array, callback, thisArg) {
+  var newArray = [];
+  for (var i = 0; i < array; i++) {
+    newArray[i] = array[i]
+  }
+  return true;
 };
 
 // Question 5
 // Write a function that returns the first element
 // for which invoking the callback returns a truthy value
-var find = function find(array, callback, thisArg) {
 
+// again, not clear what's wanted here but it looks like
+// using the properties of the .forEach method
+var find = function find(array, callback, thisArg) {
+  var result = 0;
+  if (array[i]!= null) {
+    return array[i];
```

Comment:
`i` is undefined in `find`'s scope, meaning it'll look out to its parent and ancestor scopes to find (the wrong) `i`. It implies you intended to loop, though. If this `if` statement and its body were in a loop, this function would search for the first non-null value in the array, and return it. It's not a far step from there to then test for the return value of invoking the callback on its expected arguments instead of the array at index `i` being non-null, making this function correct.

Finally, you've got unmatched curly braces. I see two opening curly braces, one for the function body and one for the `if` statement body, and only one closing curly brace below. 1/4 credit.